packages/gensx-client/src/sdk.ts

```diff
@@ -0,0 +1,370 @@
+  overrideLocalMode?: boolean; // override for devs to use locally deployed API as opposed to dev server
+}
+
+export interface RunRawOptions {
+  inputs?: Record<string, unknown>;
+  format?: "sse" | "ndjson" | "json";
+  // Allow overriding client-level config
+  org?: string;
+  project?: string;
+  environment?: string;
+}
+
+export interface StartOptions {
+  inputs?: Record<string, unknown>;
+  // Allow overriding client-level config
+  org?: string;
+  project?: string;
+  environment?: string;
+}
+
+export interface StartResponse {
+  executionId: string;
+  executionStatus: string;
+  data?: unknown;
+}
+
+export interface GetProgressOptions {
+  executionId: string;
+  format?: "sse" | "ndjson";
+}
+
+export interface WorkflowExecution {
+  executionId: string;
+  status: string;
+  progress?: unknown[];
+  result?: unknown;
+  error?: unknown;
+}
+
+/**
+ * GenSX SDK for interacting with GenSX workflows
+ *
+ * Usage:
+ * ```typescript
+ * const gensx = new GenSX({
+ *   apiKey: 'your-api-key',
+ *   org: 'your-org',
+ *   project: 'your-project',
+ *   environment: 'production'
+ * });
+ *
+ * // Start a workflow asynchronously
+ * const { executionId } = await gensx.start('workflowName', {
+ *   inputs: { userMessage: 'Hello world' }
+ * });
+ *
+ * // Or override org/project for a specific call
+ * const { executionId } = await gensx.start('workflowName', {
+ *   inputs: { userMessage: 'Hello world' },
+ *   org: 'different-org',
+ *   project: 'different-project'
+ * });
+ * ```
+ */
+export class GenSX {
+  private apiKey?: string;
+  private baseUrl: string;
+  private org?: string;
+  private project?: string;
+  private environment?: string;
+  private isLocal: boolean;
+
+  constructor(config: GenSXConfig) {
+    this.baseUrl = config.baseUrl ?? "https://api.gensx.com";
+    this.isLocal =
+      this.baseUrl.includes("localhost") && !config.overrideLocalMode;
+
+    if (!this.isLocal) {
+      // For non-local mode, require apiKey
+      this.apiKey = this.getApiKey(config.apiKey);
+      if (!this.apiKey) {
+        throw new Error(
+          "apiKey is required. Provide it in the constructor or set the GENSX_API_KEY environment variable.",
+        );
+      }
+
+      // For non-local mode, require org/project/environment
+      if (!config.org || !config.project || !config.environment) {
+        throw new Error(
+          "org, project, and environment are required when calling deployed workflows. Provide them in the constructor or method options.",
+        );
+      }
+    }
+
+    this.org = config.org;
+    this.project = config.project;
+    this.environment = config.environment;
+  }
+
+  /**
+   * Run a workflow and return the raw Response object
+   * Provides direct access to the fetch response without any processing
+   *
+   * @param workflowName - Name of the workflow to run
+   * @param options - Options including format: 'sse' | 'ndjson' | 'json'
+   * @returns Raw Response object
+   */
+  async runRaw(
```

Comment:
Would it make sense to have an easy-to-use `run` for immediately getting the progress? Or perhaps returning a listener we can bind to?

```
const execution = await gensxSdk.run(...);
execution.on('progress', (event: GenSXEvent) => {
   ...
});
```

We should consider that API for `getProgress` too.

Reply:
I think for this version I'm tempted to push as is and then we can follow up with that run capability for the sdk and layer it into the hook

Reply:
That would be really nice